### ISSUE TITLE
biom 2.1.5 fails with umlaut

### ISSUE THREAD

**Issue by johnchase** [biocore/biom-format#690]
I was attempting to work with a biom table generated by QIIME. The table was generated using the latest [UNITE ITS](https://unite.ut.ee/repository.php) database. 

biom 2.1.4 was able to work with the table, however 2.1.5 failed

The following text is contained in the file [bad_table.txt](https://raw.githubusercontent.com/johnchase/thesis_analysis/master/bad_table.txt)
[bad_table.biom](https://github.com/johnchase/thesis_analysis/blob/master/bad_table.biom?raw=true)

```
#OTU ID sample1 taxonomy
otu1    10  s__Cortinarius grosmornënsis
```

Convert this to biom

``` bash
$biom convert -i bad_table.txt -o bad_table.biom --to-hdf5 --table-type "OTU table" --process-obs-metadata taxonomy
$biom summarize-table -i bad_table.biom

Num samples: 1
Num observations: 1
Total count: 10
Table density (fraction of non-zero values): 1
```

Upgrade to biom 2.1.5

``` bash
$pip install --upgrade biom-format
$biom summarize-table -i bad_table.biom
Traceback (most recent call last):
  File "/Users/jc33/miniconda3/envs/qiime/bin/biom", line 11, in <module>
    sys.exit(cli())
  File "/Users/jc33/miniconda3/envs/qiime/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/Users/jc33/miniconda3/envs/qiime/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/Users/jc33/miniconda3/envs/qiime/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/jc33/miniconda3/envs/qiime/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/jc33/miniconda3/envs/qiime/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Users/jc33/miniconda3/envs/qiime/lib/python2.7/site-packages/biom/cli/table_summarizer.py", line 47, in summarize_table
    table = load_table(input_fp)
  File "/Users/jc33/miniconda3/envs/qiime/lib/python2.7/site-packages/biom/parse.py", line 656, in load_table
    raise TypeError("%s does not appear to be a BIOM file!" % f)
TypeError: bad_table.biom does not appear to be a BIOM file!
```

I'm not sure what the expected behavior here should be. This would cause issues outside of biom as well, however it is a common database that will likely be used with biom.

**Comment by ElDeveloper** [comment]
I don't understand how this was working before, the table contained in bad_table.txt does not have tabs to separate the columns. It's a mystery :scream: ...

**Comment by Jorge-C** [comment]
@ElDeveloper there are tabs, but you need to click to edit @johnchase's comment, and then you can copy it. Saving the edit didn't seem to work though.

**Comment by ElDeveloper** [comment]
I see, I downloaded the file linked and that one didn't have the tabs, when I add the tabs on my own, it all seems to work just fine (note how when I open on vim there's no tabs, at the end of the video I add a tab character `^I`).

![fire](https://cloud.githubusercontent.com/assets/375307/12690428/449d6752-c698-11e5-9c9a-22637ffc57b9.gif)

**Comment by Jorge-C** [comment]
Cool, @johnchase does fixing the tabs solve it for you too? I haven't been able to test on my end.

**Comment by johnchase** [comment]
Sorry when I added the original `bad_table.txt` the tab characters disappeared. I updated that file and it should be good now. I also added the converted biom table so you can test it with that. @ElDeveloper you were able to run the commands in the original post on the OTU table you created? I'm not seeing that in the video, only where you are replacing the tabs.
@Jorge-C The missing tabs were a mistake when I uploaded the file but don't have anything to do with the actual issue. Only an error uploading the file to github.

**Comment by ElDeveloper** [comment]
@johnchase, noted thanks for fixing the file. I also noticed that `biom convert` is failing likely due to the `ë` in `grosmornënsis`, here's the error I am getting:

```python
(biom-dev)yoshikivazquezbaeza:~$ biom convert -i bad_table.txt -o bad_table.biom --to-hdf5 --table-type "OTU table" --process-obs-metadata taxonomy
Traceback (most recent call last):
  File "/Users/yoshikivazquezbaeza/.virtualenvs/biom-dev/bin/biom", line 9, in <module>
    load_entry_point('biom-format==2.1.5', 'console_scripts', 'biom')()
  File "/Users/yoshikivazquezbaeza/.virtualenvs/biom-dev/lib/python2.7/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/Users/yoshikivazquezbaeza/.virtualenvs/biom-dev/lib/python2.7/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/Users/yoshikivazquezbaeza/.virtualenvs/biom-dev/lib/python2.7/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/yoshikivazquezbaeza/.virtualenvs/biom-dev/lib/python2.7/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/yoshikivazquezbaeza/.virtualenvs/biom-dev/lib/python2.7/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/Users/yoshikivazquezbaeza/.virtualenvs/biom-dev/lib/python2.7/site-packages/biom/cli/table_converter.py", line 129, in convert
    table_type, process_obs_metadata, tsv_metadata_formatter)
  File "/Users/yoshikivazquezbaeza/.virtualenvs/biom-dev/lib/python2.7/site-packages/biom/cli/table_converter.py", line 207, in _convert
    write_biom_table(result, fmt, output_filepath)
  File "/Users/yoshikivazquezbaeza/.virtualenvs/biom-dev/lib/python2.7/site-packages/biom/cli/util.py", line 35, in write_biom_table
    table.to_hdf5(f, biom.parse.generatedby())
  File "/Users/yoshikivazquezbaeza/.virtualenvs/biom-dev/lib/python2.7/site-packages/biom/table.py", line 3605, in to_hdf5
    self.group_metadata(axis='observation'), 'csr', compression)
  File "/Users/yoshikivazquezbaeza/.virtualenvs/biom-dev/lib/python2.7/site-packages/biom/table.py", line 3577, in axis_dump
    formatter[category](grp, category, md, compression)
  File "/Users/yoshikivazquezbaeza/.virtualenvs/biom-dev/lib/python2.7/site-packages/biom/table.py", line 277, in vlen_list_of_str_formatter
    data[i, :len(value)] = [v.encode('utf8') for v in value]
UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 23: ordinal not in range(128)
```

Which makes sense as it is trying to coerce that `ë` into UTF8, and it also makes sense that versions <2.1.5 were OK with this as they were not PY3K compatible.

**Comment by Jorge-C** [comment]
Thanks! I opened #691 to try and fix this (not ready yet).

**Comment by ElDeveloper** [comment]
Cool, thanks a bunch!

**Comment by wasade** [comment]
Fixed in #691
